### PR TITLE
Reenable disconnect failed resume faults

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR does three things 
* Increases ably-android version to 1.2.23
* Reenables skipped tests for `DisconnectFailedWithResume`
* Reenables skipped tests for `EnterUnresponsive`
Closes https://github.com/ably/ably-asset-tracking-android/issues/896

### CODE REVIEW

File: android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt [hunk 1]
```diff
@@ -502,7 +502,7 @@ class DisconnectWithFailedResume(apiKey: String) : ApplicationLayerFault(apiKey)
     /*
         Currently failing due to ably-java#474 presence bug
      */
-    override val skipTest = true
+    override val skipTest = false
```
Comment (AndyTWF) [inline android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt:505]: The default value of `skipTest` is false, so we should just be able to remove this completely :)

**Review by AndyTWF — CHANGES_REQUESTED** [review]

**Comment by jaley** [comment]
From my testing @ikbalkaya, it should also be safe to reenable `EnterUnresponsive`?

**Comment by ikbalkaya** [comment]
> From my testing @ikbalkaya, it should also be safe to reenable `EnterUnresponsive`?

@AndyTWF In my local, only one passes unfortunately. The one with `faultBeforeAddingTrackable` seems to fail

File: android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt (hunk 1 above)
Comment (ikbalkaya) [reply android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt]: `skipTest` now should have ceased to exist in this fault d1badac02ee6c847310b9e89c223cddec7b27387

**Review by AndyTWF — APPROVED** [review]

File: android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt (hunk 1 above)
Comment (AndyTWF) [reply android-test-common/src/main/java/com/ably/tracking/test/android/common/Faults.kt]: 👍🏻

**Comment by jaley** [comment]
OK, no worries. Let's leave it disabled if you're seeing problems and just prioritise getting the ably-java bump landed

**Comment by ikbalkaya** [comment]
@jaley I realized I had this test run without changes also from main - Now I can confirm that both test pass and I reenabled the EnterUnresponsive fault as well now

**Review by jaley — APPROVED** [review]
LGTM, thanks!

**Review by QuintinWillison — APPROVED** [review]

**Pull request merged**